File: backend/src/main/java/com/jotomo/pomo/constants/ApiConstants.java
package com.jotomo.pomo.constants;

import lombok.NoArgsConstructor;

@NoArgsConstructor
public class ApiConstants {

    public static final String BASE_PATH = "/api/v1";

    public static final String AUTH_PATH = BASE_PATH + "/auth";
    public static final String USER_PATH = BASE_PATH + "/user";
    public static final String TASK_PATH = BASE_PATH + "/task";
    public static final String PROJECT_PATH = BASE_PATH + "/project";
    public static final String SESSION_PATH = BASE_PATH + "/session";
    public static final String SESSION_CONFIGURATION_PATH = BASE_PATH + "/session-configuration";
    public static final String POMODORO = BASE_PATH + "/pomodoro";

    public static final String PATH_ID = "/{id}";
}

File: frontend/src/constants/ApiConstants.ts
export class ApiConstants {
    static readonly BASE_PATH = "http://localhost:8080/api/v1";

    static readonly AUTH_PATH = ApiConstants.BASE_PATH + "/auth";
    static readonly USER_PATH = ApiConstants.BASE_PATH + "/user";
    static readonly TASK_PATH = ApiConstants.BASE_PATH + "/task";
    static readonly PROJECT_PATH = ApiConstants.BASE_PATH + "/project";
    static readonly SESSION_PATH = ApiConstants.BASE_PATH + "/session";
    static readonly SESSION_CONFIGURATION_PATH = ApiConstants.BASE_PATH + "/session-configuration";
    static readonly POMODORO = ApiConstants.BASE_PATH + "/pomodoro";

    static readonly USER_ID_HEADER = "X-User-Id";

    static readonly PATH_ID = "/{id}";
    static readonly LOGIN = "/login";
}